docs/guides/configuring-sso-ec-deployment.md

```diff
@@ -0,0 +1,103 @@
+---
+subcategory: ""
+page_title: "Configuring an SSO provider for a deployment"
+description: |-
+    An example of how to spin up a deployment preconfigured with a SAML provider
+---
+
+# Configuring a SAML provider for an Elastic Cloud Deployment
+
+A common use case for the Elastic Cloud Terraform provider i, is to spin up an elastic deployment preconfigured with an SSO Identity provider (SAML2.0 or OIDC based)
+
+Since such a configuration often results in a cyclic dependency, where the Elasticsearch cluster yaml file requires the cluster full url name, but the url has not been generated yet, we're going to use Elastic Cloud's alias feature.
+
+*Please note, this guide doesn't go through what's needed to be configured on the IDP side, as this can change for different identity providers*
+
+First, we'll create an "random_uuid" resource, like so:
+
+```terraform
+resource "random_uuid" "uuid" {}
+```
+
+This will allow us to use a randomally generated uuid in our deployment name, so that we can later use it within the alias and revieve a unique deployment name & url alias, even before our deploymemnt is actually created. 
+
+Creating the "ec_deployment" resource, will look like this:
+
+```terraform
+resource "ec_deployment" "elastic-sso" {
+    name = format("%s-%s",var.name,substr("${random_uuid.uuid.result}",0,6))
+    alias = format("%s-%s",var.name,substr("${random_uuid.uuid.result}",0,6))
+    region = "us-east-1"
+    version = "7.17.5"
+    deployment_template_id = "aws-compute-optimized-v3"
+
+    elasticsearch {
+        topology {
+            id = "hot_content"
+            size = "8g"
+            zone_count = 2
+        }
+
+        topology {
+            id = "warm"
+            size = "8g"
+            zone_count = 2
+        }
+
+        config{
+           user_settings_yaml = templatefile("./es.yml",{kibana_url=format("https://%s-%s.kb.us-east-1.aws.found.io:9243",var.name,substr("${random_uuid.uuid.result}",0,6))})
```

Comment:
AFAICT we can't define a single domain which covers all regions which IMO makes this guide pretty painful to follow. At a minimum, we should mention "first go and create a deployment in the target region and see the format of the Kibana url's in use". 

It would be nice if we could interchangeably use `found.io`, `elastic-cloud.com`, `cloud.es.io` etc within these cluster urls. It would be even nicer if a deployment alias was a separate resource which customers could create before creating a deployment and avoid this whole palaver in the first place.

Reply:
i'll run it through fmt. I hardcoded these for sake of simplicity. this isn't about writing "correct code" its about showing how this can be achieved in the most simple form.

Reply:
@tobio it doesn't make the guide painful to follow, rather, Elastic Cloud hard to use. It's an ESS problem, that this guide won't ever solve

Reply:
I'll note the part that ESS has different URLs though just to make that bit clearer.. but I wouldn't want to introduce variables for it just for easier readability (as I'm not sharing the entire tf configuration, rather only the key parts to get this working)

Reply:
> It's an ESS problem, that this guide won't ever solve

Is there a Cloud issue to improve that situation? 

> it doesn't make the guide painful to follow

1. Create a deployment in the Cloud console in the region you want to use
2. Take a note of the format of the cluster endpoints
3. Transpose that format into the kibana_url defined here

That sounds pretty painful to me, whether we include those steps in the guide or not in reality it's what we're asking our customers to do. To be clear, I don't expect this guide to fix the underlying issue, but surely it's an issue we'd like to fix in some way? Do you know if there's an existing issue to standardise the regional domains, or implement some other solution?

Reply:
@tobio it's an infra issue for the most part, i'd ask @yuvielastic